### DISCUSSION TITLE
Debugging OrchardCore module without restarting whole site?

### DISCUSSION THREAD

**Discussion by MRSheikh** [OrchardCMS/OrchardCore#8493] [Ideas]
hi
i have large site and write orchardcore module for it and it's work fine. but whenever i change my client java-script code or HTML code of my module , changes not affected until i restart visual studio debugging mode. is there a way to live debug my orchard core module even my site is running?

**Comment by jtkech** [comment]
Set the `ASPNETCORE_ENVIRONMENT` variable to `Development`

We have custom static files providers that grabs static files under the module project locations if in dev mode, this in place of serving the static ones that are embedded in the modules assemblies in prod mode

**Reply by MRSheikh** [reply]
thanks. i check my site DEBUG properties: 
![image](https://user-images.githubusercontent.com/17869166/106716977-4fda0780-6614-11eb-9fbe-94c7ce9ea1ba.png)

and my module DEBUG properties: 
![image](https://user-images.githubusercontent.com/17869166/106717056-6aac7c00-6614-11eb-850d-8a15643d70c8.png)

both have `ASPNETCORE_ENVIRONMENT` variable but nothing happened .

**Reply by jtkech** [reply]
Under which folder is your static file, need to be under the `wwwroot` of your module, check under Visual Studio if it is marked as embedded resource, it needs to be embeeded to be found, even we serve it from the project location in dev mode, check in your module project file that here is no item group removing it from the embedded resources

**Reply by MRSheikh** [reply]
i changed .csproj file and add this:
`<ItemGroup>
    <Content Include="Views\**" />
    <Content Include="wwwroot\**" />
  </ItemGroup>`

but when i change javascript code of my view and save it, nothing happen on view.

**Reply by jtkech** [reply]
@MRSheikh About razor views runtime compilation

I assume it is a project file of a **module**, so remove this item group, module views still need to be embedded, this is done automatically, you just have to remove this extra item group

Then it depends if you are using the full source code (a little more complex), or if you are referencing `OrchardCore` packages, the simpler way is to show your module project file.

At least you need to use in your main app project file the same property, as we use in our `OrchardCore.Cms.Web.csproj`

    <PreserveCompilationReferences>true</PreserveCompilationReferences>

If you use `OrchardCore` as packages, normally the above and the env variable we talked about are sufficient

**Reply by MRSheikh** [reply]
> <PreserveCompilationReferences>true</PreserveCompilationReferences>

thanks. that works fine !!!

**Reply by dafergu2** [reply]
Just as a side note, I've had good luck with using https://www.livesharp.net/ in conjunction with modules and themes.  It's really quite nice.

**Comment by MRSheikh** [comment]
**Setting** 

`<PropertyGroup> 
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <PreserveCompilationReferences>true</PreserveCompilationReferences>
  </PropertyGroup>`

**on main site project that referenced `OrchardCore.Application.Cms.Targets.dll` works fine !!!**
